thirtyinch/src/main/java/net/grandcentrix/thirtyinch/distinctuntilchanged/DistinctUntilChangedInvocationHandler.java

```diff
@@ -28,15 +29,15 @@
 
     private static final String TAG = DistinctUntilChangedInvocationHandler.class.getSimpleName();
 
-    private HashMap<String, Integer> mLatestMethodCalls = new HashMap<>();
+    private HashMap<String, WeakReference<Object[]>> mLatestMethodCalls = new HashMap<>();
```

Comment:
restating my deleted comment:

Since `WeakHashMap` treats keys as objects that can be garbage collected, and here we use method names as keys and method arguments as values, I don't think using `WeakHashMap ` will be a gain here